src/sql/src/plan/lowering/variadic_left.rs

```diff
@@ -389,6 +399,19 @@ fn decompose_equations(predicate: &MirScalarExpr) -> Option<Vec<(usize, usize)>>
             _ => return None,
         }
     }
+
+    // Remove duplicates
```

Comment:
This extension of `decompose_equations` should fix the issue (the regression test doesn't fail). Once I have a release build I will also try RQG.